handle/handle_test.go

```diff
@@ -337,6 +337,55 @@ func TestIgnoreIndex(t *testing.T) {
 	}
 }
 
+func TestPreventListings(t *testing.T) {
+	testCases := []struct {
+		name     string
+		path     string
+		code     int
+		contents string
+	}{
+		{"Good base dir", "", ok, tmpIndex},
+		{"Good base index", tmpIndexName, redirect, nothing},
+		{"Good base file", tmpFileName, ok, tmpFile},
+		{"Bad base file", tmpBadName, missing, notFound},
+		{"Good subdir dir", subDir, ok, tmpSubIndex},
+		{"Good subdir index", tmpSubIndexName, redirect, nothing},
+		{"Good subdir file", tmpSubFileName, ok, tmpSubFile},
```

Comment:
For 100% coverage, can you add:

1. In the variables at the top of `handle_test.go`, right after the assignment of `tmpSubDeepBadName` variables for `tmpNoIndexDir` and `tmpNoIndexName`:

```go
var (
	baseDir             = "tmp/"
	...
	tmpSubDeepBadName   = "sub/deep/bad.txt"
	tmpNoIndexDir       = "noindex/"                     <--
	tmpNoIndexName      = "noindex/noindex.txt"          <--
```

Then add `tmpNoIndexName` to files (can re-use another file's contents, as shown):

```go
files = map[string]string{
    baseDir + tmpIndexName:        tmpIndex,
    ...
    baseDir + tmpSubDeepFileName:  tmpSubDeepFile,
    baseDir + tmpNoIndexName:      tmpSubDeepFile,       <--
}
```

Which then makes it possible to test the case where there is a directory without an `index.html` file:

```go
{"Good subdir file", tmpSubFileName, ok, tmpSubFile},
{"Dir without index", tmpNoIndexDir, missing, notFound},  <--
```